fuel-txpool/src/service.rs

```diff
@@ -179,11 +181,9 @@ impl Context {
 
                     tokio::spawn( async move {
                         let txpool = txpool.as_ref();
-                        match new_transaction.unwrap() {
-                            TransactionBroadcast::NewTransaction ( tx ) => {
-                                let txs = vec!(Arc::new(tx));
-                                TxPool::insert(txpool, db.as_ref().as_ref(), tx_status_sender, txs).await
-                            }
+                        if let GossipData { data: Some(TransactionBroadcast::NewTransaction ( tx )), .. } =  new_transaction.unwrap() {
```

Comment:
This is the unwrap that I mean

Reply:
that `unwrap` is from `Result<TransactionGossipData>` not from the `Option` if that's what you meant